src/qtiItem/helper/simpleParser.js

```diff
@@ -102,40 +103,55 @@ function buildTooltip(targetHtml, contentId, contentHtml) {
     };
 }
 
+function buildTable($elt) {
+    var elt = buildElement($elt);
+
+    //set table body
+    elt.body = {
+        body: $elt.html(),
+        elements: {}
+    };
+    return elt;
+}
+
 function parseContainer($container, options) {
     var ret = {
         serial: util.buildSerial('_container_'),
         body: '',
         elements: {}
     };
 
-    _.each(_parsableElements, function(qtiClass) {
-        $container.find(qtiClass).each(function() {
+    _.each(_parsableElements, function (qtiClass) {
+        $container.find(qtiClass).each(function () {
             var $qtiElement = $(this);
             var element = buildElement($qtiElement, options);
 
+            // rendering the table
+            if (qtiClass === _defaultOptions.ns.table) {
+                element = buildTable($qtiElement, options);
```

Comment:
Please, optimise code. In the case with `table` you call `buildElement` twice

Reply:
code optimized and changed to ES6- for fail safe , i use most of them as let and some of them are in const